### DISCUSSION TITLE
New escape character

### DISCUSSION THREAD

**Discussion by protofolius** [comictagger/comictagger#703] [Q&A] [answered]
In the old syntax , the escape character was ^ . But I can not seem to find what is the escape character in the new syntax . ^ does not seem to work anymore .

For example I want to CLI tag the following :

- -m "imprint: 'Mr. d'Hoore-Achel', title: '1=3'"

**Comment by lordwelch** [comment]
It uses yaml with implied `{}` added if it doesn't start with them, yaml doesn't need quotes
`-m "imprint: Mr. d'Hoore-Achel, title: 1=3"` == `-m "{imprint: Mr. d'Hoore-Achel, title: 1=3}"`
== `-m "{imprint: 'Mr. d''Hoore-Achel', title: '1=3'}"`

yaml also supports json so you can do 

`-m '{"imprint": "Mr. d'Hoore-Achel", "title": "1=3"}'`

**Comment by protofolius** [comment]
I think I understand it somewhat better now . Yet I am still struggling to find the proper syntax when I need to tag multiple persons for imprint , or an extended description . I feel like some characters need to be escaped , or at least handled differently .

2 examples of imprint and description that I do not get properly syntaxed

imprint: Mr. d'Hoore-Achel, Mrs. Van Dijk,

description: Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and desctruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney.

**Reply by lordwelch** [reply]
Assuming bash
```bash
comictagger -m "imprint: 'Mr. d''Hoore-Achel, Mrs. Van Dijk'," 

comictagger -m "description: \" Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney.\""
```

Also imprint doesn't usually make sense to have multiple even less so with a person's name

**Reply by protofolius** [reply]
Thank you for helping lordwelch !! Much appreciated .
Yet , I excuse myself for the nitpicking , but since that code still would not let me tag , I think some small errors were in the commands .
Could it be that this is the corrected version of what you were saying ?

(, corrected)
`comictagger -m "imprint: 'Mr. d''Hoore-Achel,Mrs. Van Dijk'" `


(a few ' added , starting space removed, double single quotes added for Junglebook)
`comictagger -m "description: '\"Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the ''Junglebook'' stories of Kipling and Walt-Disney.\"'"`


At least in this way , it lets me tag ! Let me know if things would still not be according to the syntax rules .

**Reply by lordwelch** [reply]
It sounds like you aren't using bash. What shell are you using? (a)sh, bash, zsh, fish, xonsh, pwsh (powershell), csh, ksh, cmd.exe

There are two parts to it the first part is yaml, but since you are typing it into a commandline the commandline also needs to parse it.
This is valid yaml:
```yaml
description: " Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney."
```
but when you type it into a shell it ends up as two arguments
```plain
description:
```
and
```plain
" Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney."
```

You can see this when using bash by using the `printf` command:
```console
bash-5.2$ printf '%s\n' description: " Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney."
description:
 Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney.
```
So to make it so that it reads it properly you need to quote it again
```console
bash-5.2$ printf '%s\n' "description: \" Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney.\""
description: " Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the 'Junglebook' stories of Kipling and Walt-Disney."
```

Your second command is not quite correct as it has extra quotes
```console
bash-5.2$ printf '%s\n' "description: '\"Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the ''Junglebook'' stories of Kipling and Walt-Disney.\"'"
description: '"Mowgli is his name, a.k.a The Kid. He was born when the big war started. He lived in a world of chaos, hate and destruction. A world where clan leaders crowned themselves king of small member states. It is in this jungle where The Kid tries to survive together with his educator Baloo and negress Begheera. The Kid is a wink towards the ''Junglebook'' stories of Kipling and Walt-Disney."'
```

Both of the commands I gave are correct they will let you tag unless you are on a different shell as they will have different quoting rules

**Reply by protofolius** [reply]
Well , I am creating a Python script which I run in a Windows program called Pulsar . When I run the script , it opens a black screen , which I think is a DOS terminal .

With the help of your examples I managed to fix some of the things . There is one thing however that I have not been able to figure out and that is how to escape a whole block entirely , like you did on Bash with` \" block of escaped text \"` . It would be very helpful for the description field .

**Reply by lordwelch** [reply, marked as answer]
If you didn't figure things out yet.
One problem is that [windows isn't consistent][fuck windows] on splitting commandline arguments. So I did make it so that you can put an `@` in front and it will treat it as a file to read so you can do `-m "@metadata.yaml"` and then you don't need to worry about escaping you can just put it in the file directly

[fuck windows]: https://learn.microsoft.com/en-us/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way
